src/Medidata.ZipkinTracer.Core/TraceProvider.cs

```diff
@@ -72,11 +73,11 @@ internal TraceProvider(IZipkinConfig config, IOwinContext context = null)
                 headerSampled = context.Request.Headers[SampledHeaderName];
             }
 
-            TraceId = Parse(headerTraceId) ? headerTraceId : GenerateHexEncodedInt64FromNewGuid();
-            SpanId = Parse(headerSpanId) ? headerSpanId : TraceId;
-            ParentSpanId = Parse(headerParentSpanId) ? headerParentSpanId : string.Empty;
+            TraceId = headerTraceId.IsParsableTo128Or64Bit() ? headerTraceId : GenerateHexFromNewGuid(config.Create128BitTraceId);
```

Comment:
one way out is to have a flag as to whether to break or restart on malformed ID. For example, in spring-cloud-sleuth we rolled back breaking on propagation error (vs restarting the trace) because the middleware that had the bug couldn't be updated with a fix fast enough and breaking caused production failures. Such bugs are rare, though, so not necessarily the best answer for everyone.